### DISCUSSION TITLE
Is it possible to enable SSG but skipping the first build ?

### DISCUSSION THREAD

**Discussion by dotlouis** [vercel/vercel#5466] [Help] [answered]
Hello,
I'm working on an app with user-generated pages.
This means I could potentially have many many pages on my NextJS project.
For now, I don't have a lot so I maybe am jumping the gun a bit there.
My question is:
> Is it possible to have pages generated statically (SSG with `getStaticProps`) without having them all generated on the first build (with `getStaticPaths`)?

Having to generate all the pages on the first build could potentially take a long time whereas a large chunk of the pages would not be accessed right away.
To summarize, the ideal flow to me would like this:
1. Have the first build generate no static pages except for a few of them I know in advance. It's fast.
2. Upon page request, have SSG kick in and generate only the requested static page

Thanks in advance

**Comment by matthewlilley** [comment, marked as answer]
@dotlouis You can use the fallback option which means pages will be rendered on-demand for the first time and return an app shell while router.isFallback is true. I haven't done this for the majority of pages, but think I will shortly because builds are also becoming a pain.

https://nextjs.org/docs/basic-features/data-fetching#fallback-false

Good luck

**Reply by dimroth** [reply]
@matthewlilley it won't affect the first build as mentioned in the docs: 

> The paths returned from getStaticPaths will be rendered to HTML at build time by getStaticProps.

**Reply by coetry** [reply]
@dimroth if you don't return anything (empty array) from `getStaticPaths`, then nothing will be built initially and everything will be built on-demand.

**Reply by dimroth** [reply]
@coetry I don't know how I didn't think about that but it worked! I am sure it is worth mentioning in the documentation with an example [here](https://nextjs.org/docs/basic-features/data-fetching#when-is-fallback-true-useful) 😇
